$NetBSD: patch-configure,v 1.3 2025/04/21 20:54:56 wiz Exp $

* Avoid bash specific variable substitution.
* Ignore doxygen.
* Ignore VCS.
* Handle SSP in pkgsrc.
* Retain _XOPEN_SOURCE, fixes eventmachine.
* Fix argument for pthread_self() on NetBSD.
* Explictly stop display with color.
* Remove bogus MJIT disabling on SunOS.

--- configure.orig	2022-12-25 06:55:27.000000000 +0000
+++ configure
@@ -8666,7 +8666,7 @@ else $as_nop
 then :
 
 	case "$target_os" in #(
-  darwin*) :
+  nodarwin*) :
 
 	    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for real target cpu" >&5
 printf %s "checking for real target cpu... " >&6; }
@@ -10720,6 +10720,7 @@ fi
 	stack_protector=no
      ;; #(
   *) :
+	stack_protector=no
      ;;
 esac
     if test -z "${stack_protector+set}"
@@ -12607,18 +12608,6 @@ then :
      CPPFLAGS="$CPPFLAGS ${rb_opt}" ;;
 esac
 	done
-		    # _XOPEN_SOURCE should not be defined for C++ on Solaris.
-		    # RUBY_APPEND_OPTIONS(CXXFLAGS)
-	for rb_opt in -U_XOPEN_SOURCE; do
-	case " ${CXXFLAGS-} " in #(
-  *" ${rb_opt} "*) :
-     ;; #(
-  '  ') :
-     CXXFLAGS="${rb_opt}" ;; #(
-  *) :
-     CXXFLAGS="$CXXFLAGS ${rb_opt}" ;;
-esac
-	done
 
 fi
 
@@ -25301,7 +25294,9 @@ else
 #include <stdlib.h>
 #include <stddef.h>
 #ifndef alloca
-# ifdef __GNUC__
+# if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__DragonFly__) || defined(__OpenBSD__)
+#   include <stdlib.h>
+# elif defined __GNUC__
 #  define alloca __builtin_alloca
 # elif defined _MSC_VER
 #  include <malloc.h>
@@ -31638,8 +31635,10 @@ fi
 	 ;; #(
   freebsd*|dragonfly*) :
 
+	RUBY_SO_NAME="${RUBY_SO_NAME}"'.$(RUBY_PROGRAM_VERSION)'
 	LIBRUBY_SO='lib$(RUBY_SO_NAME).$(SOEXT).$(MAJOR)$(MINOR)'
-	LIBRUBY_SONAME='$(LIBRUBY_SO)'
+	LIBRUBY_SONAME='lib$(RUBY_BASE_NAME).$(RUBY_API_VERSION).$(SOEXT)'
+	LIBRUBY_ALIASES='$(LIBRUBY_SONAME) lib$(RUBY_INSTALL_NAME).$(SOEXT)'
 	if test "$rb_cv_binary_elf" != "yes"
 then :
 
@@ -31732,9 +31731,10 @@ esac
 	 ;; #(
   darwin*) :
 
+	RUBY_SO_NAME="${RUBY_SO_NAME}"'.$(RUBY_PROGRAM_VERSION)'
 	LIBRUBY_SO='lib$(RUBY_SO_NAME).$(SOEXT)'
-	LIBRUBY_SONAME='$(LIBRUBY_SO)'
-	LIBRUBY_ALIASES='lib$(RUBY_INSTALL_NAME).$(SOEXT)'
+	LIBRUBY_SONAME='lib$(RUBY_BASE_NAME).$(RUBY_API_VERSION).$(SOEXT)'
+	LIBRUBY_ALIASES='$(LIBRUBY_SONAME) lib$(RUBY_INSTALL_NAME).$(SOEXT)'
 	if test "$load_relative" = yes
 then :
 
@@ -31950,7 +31955,7 @@ then :
 esac
     rpathflag=`IFS="$PATH_SEPARATOR"
         echo x "$rpathflag" |
-        sed "s/^x *//;s${IFS}"'%1\\$-s'"${IFS}${libprefix}${IFS}g;s${IFS}%s${IFS}${libprefix}${IFS}g"
+        sed "s/^x *//;s${IFS}"'%1\\$-s'"${IFS}${libprefix}${IFS}g;s${IFS}%s${IFS}${PREFIX}/lib${IFS}g"
     `
     LIBRUBY_RPATHFLAGS="$LIBRUBY_RPATHFLAGS${rpathflag}"
     LIBRUBYARG_SHARED="$LIBRUBY_RPATHFLAGS $LIBRUBYARG_SHARED"
@@ -32603,7 +32608,7 @@ then :
   enableval=$enable_jit_support; MJIT_SUPPORT=$enableval
 else $as_nop
   case "$target_os" in #(
-  wasi | mingw* | solaris*) :
+  wasi | mingw*) :
     MJIT_SUPPORT=no ;; #(
   *) :
     MJIT_SUPPORT=yes
@@ -33885,7 +33890,7 @@ guard=INCLUDE_RUBY_CONFIG_H
 (
   if test "x$CONFIGURE_TTY" = xyes
 then :
-  color=--color
+  color=--color=never
 else $as_nop
   color=
 fi
@@ -35124,21 +35131,7 @@ which seems to be undefined.  Please mak
     "Makefile":F)
     tmpmk=confmk$$.tmp
     {
-	if test ${VCS+set}
-then :
-
-	    :
-
-elif git_dir=`$GIT --work-tree="$srcdir" --git-dir="$srcdir/.git" rev-parse --git-dir 2>/dev/null`
-then :
-
-	    VCS='$(GIT)'
-
-else $as_nop
-
 	    VCS='echo cannot'
-
-fi
 	case "$VCS" in #(
   '$(GIT)'|git) :
     VCSUP='$(VCS) pull --rebase $(GITPULLOPTIONS)' ;; #(
